web/src/installers/installer-versions.ts

```diff
@@ -4,14 +4,41 @@ import Bugsnag from "@bugsnag/js";
 import { HTTPError } from "../server/errors";
 import { getPackageUrl } from "../util/package";
 import { InstallerVersions } from "./versions";
+import * as semver from "semver";
+import {logger} from "../logger";
 
 interface IInstallerVersions {
   [addon: string]: string[];
 }
 
 const installerVersionsCache: { [url: string]: IInstallerVersions } = {};
+let externalAddons = {};
+let externalAddonTimer;
 
-export async function getInstallerVersions(distUrl: string, kurlVersion?: string): Promise<IInstallerVersions> {
+function mergeAddonVersions(internalAddonVersions: IInstallerVersions, kurlVersion?: string) {
+  const addons: IInstallerVersions = {};
+  Object.keys(externalAddons).forEach(externalAddonName => {
+    const fileName = externalAddonName.slice(0, externalAddonName.length - 7);
```

Comment:
What is the significance of magic number `7` here? Can we make that a constant somewhere so it's a bit more self-documenting?

Reply:
Stripping off ".tar.gz", I could either make a constant or a function to strip all file extensions.  Node's path API has functionality to get file name without extension but it considers the file name for "test.tar.gz" to be "test.tar"